operations_on_datastructures/array_left_rotation_efficient.cpp

```diff
@@ -0,0 +1,64 @@
+#include <iostream>
+/**
+ * This program can be used to rotate an array d times towards left.
+ * Variables used:
+ * SIZE         : Size of the Array.
+ * no_of_rotate : Number of indices you want to rotate.
+ * arr          : Used to copy first no_of_rotate elements of the array.
+ * a            : Main array used.
+ * i            : Loop variable.
+ * j            : Loop variable.
+ *
+ * This program uses a more efficient logic to rotate the array as each element
+ * is roated d times in only single iteration. This uses less time for more long
+ * arrays.
+ *
+ * The Algorithm used is:
+ * 1) Input array, size, and no of times to rotate.
+ * 2) Create a new array and save first no_of_rotate elements into it.
+ * 3) Copy the remaining elements from starting index of the input array.
+ * 4) Copy back the elements saved in the new array to the input array.
+ * 5) Print the array. */
+
+void rotateArray(int *a, int SIZE, int no_of_rotate) { /** Rotating the Array */
+  int *arr = new int[no_of_rotate];
+  for (int i = 0; i < no_of_rotate; i++) {
```

Comment:
Could just use `memcpy` Instead of a loop

Reply:
It will be better to understand and learn with loops, I guess.

Reply:
refer to this for help 
http://www.cplusplus.com/reference/cstring/memcpy/ 
https://en.cppreference.com/w/cpp/string/byte/memcpy
if required

Reply:
i think it would be better to keep both things just comment out the loops in the code and add the `memcpy()` implementation also put a NOTE there telling a bit about both the implementaions, i think this would help people to learn and understand in better way